internal/goofys.go

```diff
@@ -975,11 +986,14 @@ func (fs *Goofys) CreateFile(
 
 	inode, fh := parent.Create(op.Name)
 
+	parent.mu.Lock()
+
 	fs.mu.Lock()
 	defer fs.mu.Unlock()
-
 	fs.insertInode(parent, inode)
```

Comment:
insertInode() has the following comments but we were not acquiring the parent lock before we call.

// LOCKS_REQUIRED(fs.mu)
// LOCKS_REQUIRED(parent.mu)